eng/scripts/ci-source-build.sh

```diff
@@ -9,10 +9,10 @@ set -euo pipefail
 scriptroot="$( cd -P "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 reporoot="$(dirname "$(dirname "$scriptroot")")"
 
-export DotNetBuildFromSource='true'
-
- # Build repo tasks
+# Build repo tasks
 "$reporoot/eng/common/build.sh" --restore --build --ci --configuration Release /p:ProjectToBuild=$reporoot/eng/tools/RepoTasks/RepoTasks.csproj
 
- # Build projects
+export DotNetBuildFromSource='true'
```

Comment:
Turns out that installing the runtime is explicitly disabled in source build in Arcade. The reason is that the runtime is expected to be built during source-build. However, in our "fake" source build, it seems like we need to install the runtime. I'm modifying the behaviour so that the runtime is installed during repo task restore to unblock this.

Curious why this changed though. What's changed in rzc?